BuildScript.Nuke/GenerateSbom/SolutionSbomGenerator.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership.  rubicon licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.  You may obtain a copy of the
+// License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Nuke.Common.IO;
+using Nuke.Common.ProjectModel;
+using Nuke.Common.Tooling;
+using Nuke.Common.Tools.Npm;
+using Nuke.Common.Tools.PowerShell;
+using SbomCleaner.Library;
+using Serilog;
+
+namespace Remotion.BuildScript.GenerateSbom;
+
+public class SolutionSbomGenerator: ISbomGenerator
+{
+  public SbomGeneration SbomGeneration => SbomGeneration.ForSolution;
+
+  private readonly FilterList _packageBlackList;
+  private readonly FilterList _projectBlacklist;
+
+  private readonly string _githubUsername;
+  private readonly string _githubAccessToken;
+
+  private readonly AbsolutePath _outputFile;
+
+  private readonly Solution _solution;
+  private readonly AbsolutePath _workingDirectory;
+
+  private readonly AbsolutePath? _pathToPackageJson;
+
+  public SolutionSbomGenerator (
+      Solution solution,
+      AbsolutePath workingDirectory,
+      AbsolutePath outputFile,
+      FilterList packageBlackList,
+      FilterList projectBlacklist,
+      AbsolutePath? pathToPackageJson,
+      string githubUsername,
+      string githubAccessToken)
+  {
+    _solution = solution;
+    _workingDirectory = workingDirectory;
+    _outputFile = outputFile;
+    _projectBlacklist = projectBlacklist;
+    _packageBlackList = packageBlackList;
+    _pathToPackageJson = pathToPackageJson;
+    _githubUsername = githubUsername;
+    _githubAccessToken = githubAccessToken;
+  }
+
+  public async Task Generate (Tool cycloneDxTool)
+  {
+    var uncleanedSbomFilename = "uncleaned-sbom.xml";
+
+    Directory.CreateDirectory(_workingDirectory);
+
+    cycloneDxTool.Invoke(
+        arguments:
+        $"-o . --disable-package-restore --exclude-dev --exclude-test-projects -gu {_githubUsername} -gt {_githubAccessToken} -f \"{uncleanedSbomFilename}\" {_solution.Path}",
+        workingDirectory: _workingDirectory
+    );
+
+    Log.Information("Cleaning Sbom...");
```

Comment:
we could write `SBOM` or `sbom` instead of `Sbom` in the various log messages. Pascal case isn't really common for this type of thing and SBOM isn't any different from JSON in the log message.